Assets/SO Architecture/Editor/ClampValueHelper.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using UnityEditor;
+using UnityEngine;
+
+#if ODIN_INSPECTOR
+using Sirenix.Utilities.Editor;
+#endif
+
+namespace ScriptableObjectArchitecture.Editor
+{
+    /// <summary>
+    /// A helper class for drawing better controls for clamped value types (mostly simple C# types)
+    /// </summary>
+    public static class ClampValueHelper
+    {
+        private static readonly List<Type> SUPPORTED_INT_TYPES;
+        private static readonly List<Type> SUPPORTED_FLOAT_TYPES;
+
+        static ClampValueHelper()
+        {
+            SUPPORTED_INT_TYPES = new List<Type>
+            {
+                typeof(sbyte),
+                typeof(byte),
+                typeof(short),
+                typeof(ushort),
+                typeof(int)
+            };
+
+            SUPPORTED_FLOAT_TYPES = new List<Type>
+            {
+                typeof(float)
+            };
+        }
+
+        /// <summary>
+        ///  Draws a slider for the passed <see cref="SerializedProperty"/> values.
+        /// </summary>
+        /// <param name="valueProp"></param>
+        /// <param name="minProp"></param>
+        /// <param name="maxProp"></param>
+        public static void DrawClampRangeLayout(
+            SerializedProperty valueProp,
+            SerializedProperty minProp,
+            SerializedProperty maxProp)
+        {
+            var valueType = GetType(valueProp);
+
+            if (SUPPORTED_INT_TYPES.Contains(valueType))
+            {
+                #if ODIN_INSPECTOR
+                valueProp.intValue = SirenixEditorFields.RangeIntField(
+                    new GUIContent(valueProp.displayName),
+                    valueProp.intValue,
+                    minProp.intValue,
+                    maxProp.intValue);
+                #else
+                EditorGUILayout.IntSlider(valueProp, minProp.intValue, maxProp.intValue);
+                #endif
+            }
+            else if (SUPPORTED_FLOAT_TYPES.Contains(valueType))
+            {
+                #if ODIN_INSPECTOR
+                valueProp.floatValue = SirenixEditorFields.RangeFloatField(
+                    new GUIContent(valueProp.displayName),
+                    valueProp.floatValue,
+                    minProp.floatValue,
+                    maxProp.floatValue);
+                #else
+                EditorGUILayout.Slider(valueProp, minProp.floatValue, maxProp.floatValue);
+                #endif
+            }
+        }
+
+        /// <summary>
+        /// Returns true if the passed <paramref name="valueProp"/> can be drawn with better clamp controls.
+        /// </summary>
+        /// <param name="valueProp"></param>
+        /// <returns></returns>

```

Comment:
The <returns></returns> comments get added automatically, and have to be removed manually.  It's tempting, isn't it?  But I won't insist.

Reply:
I don't generally ask for these to be added/removed when VS adds them automatically.